circularBuffer.hpp

```diff
@@ -0,0 +1,116 @@
+#ifndef circularBuffer_
+#define circsularBuffer_
+
+#define BufferSize 5             //circular buffer size 
```

Comment:
Why not a template argument?

Reply:
Yes could be done like:

template <class T,uint32_t BufferSize>

Reply:
Yes can be done like that. 

template <class T, uint32_t BufferSize> class circularBuffer

I must refresh my C++ knowledge.

Reply:
👍